ext/msgpack/packer.c

```diff
@@ -132,6 +166,10 @@ bool msgpack_packer_try_write_with_ext_type_lookup(msgpack_packer_t* pk, VALUE v
             pk->buffer = parent_buffer;
             msgpack_packer_write_ext(pk, ext_type, payload);
         }
+
+        if (mapped_strings_count > 0) {
+            RB_GC_GUARD(mapped_strings[0]);
+        }
```

Comment:
I'm not sure what the purpose of this `RB_GC_GUARD` is. AFAIK the compiler isn't allowed to reuse the memory allocated in `alloca`.

Reply:
Just me having no idea. I'll remove it.

Reply:
So it turns out clang does optimize the alloca away :/. I'll add this back for now.